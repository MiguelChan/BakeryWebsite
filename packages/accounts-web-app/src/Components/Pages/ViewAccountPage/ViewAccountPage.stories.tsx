import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { AccountType } from '@mgl/shared-components';
import { ViewAccountPage } from './ViewAccountPage';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { UseGetAccountState } from '../../../Hooks';

export default {
  title: 'Components/Pages/ViewAccountPage',
  component: ViewAccountPage,
} as Meta;

interface Props {
  appContext: ApplicationContext;
}

const Template: Story<Props> = ({ appContext }) => (
  <AccountsAppContext.Provider value={appContext}>
    <ViewAccountPage />
  </AccountsAppContext.Provider>
);

export const Primary = Template.bind({});
Primary.args = {
  appContext: {
    useGetAccount: (): UseGetAccountState => ({
      account: null,
      getAccount: (): void => {},
      isLoading: false,
      errorMessage: undefined,
    }),
  },
};

export const Loading = Template.bind({});
Loading.args = {
  appContext: {
    useGetAccount: (): UseGetAccountState => ({
      account: null,
      getAccount: (): void => {},
      isLoading: true,
      errorMessage: undefined,
    }),
  },
};

export const WithAccount = Template.bind({});
WithAccount.args = {
  appContext: {
    useGetAccount: (): UseGetAccountState => ({
      account: {
        accountType: AccountType.Expenses,
        id: 'SomeSome',
        subAccounts: [{
          description: 'SomeSomeDescription',
          id: 'SomeSeme',
        }],
        title: 'AnAcct',
      },
      getAccount: (): void => {},
      isLoading: false,
      errorMessage: undefined,
    }),
  },
};

export const WithErrorMessage = Template.bind({});
WithErrorMessage.args = {
  appContext: {
    useGetAccount: (): UseGetAccountState => ({
      account: {
        accountType: AccountType.Expenses,
        id: 'SomeSome',
        subAccounts: [{
          description: 'SomeSomeDescription',
          id: 'SomeSeme',
        }],
        title: 'AnAcct',
      },
      getAccount: (): void => {},
      isLoading: false,
      errorMessage: 'Hay un error intermitente',
    }),
  },
};
