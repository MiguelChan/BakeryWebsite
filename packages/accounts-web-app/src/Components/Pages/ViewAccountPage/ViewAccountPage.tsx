import { isNullOrUndefined } from '@mgl/shared-components';
import {
  Alert,
  AlertTitle,
  Typography,
} from '@mui/material';
import React from 'react';
import {
  useParams,
} from 'react-router-dom';
import {
  AccountsAppContext,
  ApplicationContext,
} from '../../../Context';
import { LoadingDialog } from '../../Blocks';
import { EditableAccountForm } from '../../Constructed';
import { SimpleAppTemplate } from '../../Templates';

export interface ViewAccountPageProps {

}

export const ViewAccountPage: React.FunctionComponent<ViewAccountPageProps> = () => {
  const {
    accountId,
  } = useParams<{ accountId: string }>();

  const {
    useGetAccount,
  } = React.useContext<ApplicationContext>(AccountsAppContext);
  const useGetAccountState = useGetAccount();

  React.useEffect(() => {
    useGetAccountState.getAccount(accountId);
  }, []);

  const renderBody = (): React.ReactElement => {
    if (useGetAccountState.isLoading) {
      return (
        <LoadingDialog onClose={null} />
      );
    }

    if (isNullOrUndefined(useGetAccountState.account)) {
      return (
        <Alert severity='error'>
          <AlertTitle>
            Oops!
          </AlertTitle>
          Ha habido un <strong>error</strong> al momento de intentar de cargar la cuenta con id.
          Favor de intentar mas tarde
        </Alert>
      );
    }

    return (
      <>
        {useGetAccountState.errorMessage && <Alert severity='error'>{useGetAccountState.errorMessage}</Alert>}
        <EditableAccountForm
          account={useGetAccountState.account}
          onSubmitAccount={(): void => {}}
          title='Vista de Cuenta'
          readOnly
        />
      </>
    );
  };

  return (
    <SimpleAppTemplate renderBody={renderBody} />
  );
};
