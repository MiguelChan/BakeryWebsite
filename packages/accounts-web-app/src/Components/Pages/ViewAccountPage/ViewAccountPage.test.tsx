import React from 'react';
import {
  RenderResult,
  render,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { ApplicationContext } from '../../..';
import { AccountsAppContext } from '../../../Context';
import { ViewAccountPage } from './ViewAccountPage';
import {
  Loading,
  Primary,
  WithAccount,
  WithErrorMessage,
} from './ViewAccountPage.stories';

interface Props {
  appContext: ApplicationContext;
}

describe('ViewAccountPage', () => {
  const setupComponent = ({ appContext }): RenderResult => {
    const Component = (
      <AccountsAppContext.Provider value={appContext}>
        <MemoryRouter>
          <ViewAccountPage />
        </MemoryRouter>
      </AccountsAppContext.Provider>
    );

    return render(Component);
  };

  it('Should render an Error dialog when the Account cant be found', () => {
    const renderResult = setupComponent(Primary.args);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should render an Error dialog when an error exists', () => {
    const renderResult = setupComponent(WithErrorMessage.args);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should render a Loading dialog', () => {
    const renderResult = setupComponent(Loading.args);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should render the Account', () => {
    const renderResult = setupComponent(WithAccount.args);
    expect(renderResult.container).toMatchSnapshot();
  });
});
