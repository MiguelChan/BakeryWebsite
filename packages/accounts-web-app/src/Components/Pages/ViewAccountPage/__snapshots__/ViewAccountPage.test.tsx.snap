// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAccountPage Should render a Loading dialog 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  />
</div>
`;

exports[`ViewAccountPage Should render an Error dialog when an error exists 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-145dpyk-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        Hay un error intermitente
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h5 css-1hxkzy9-MuiTypography-root"
      >
        Vista de Cuenta
      </h4>
      <form
        id="create-account"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="title"
                id="title-label"
              >
                Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="AccountTitleId"
                  disabled=""
                  id="title"
                  name="title"
                  type="text"
                  value="AnAcct"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Tipo de Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-disabled="true"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="accountType"
                  class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="accountType"
                  role="button"
                >
                  Gastos
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  disabled=""
                  name="accountType"
                  tabindex="-1"
                  value="Expenses"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Tipo de Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Subcuentas
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Id de Subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Nombre de subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="description"
                        id="description-label"
                      >
                        Descripcion Subcuenta
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          data-testid="ATextField"
                          disabled=""
                          id="description"
                          name="description"
                          type="text"
                          value="SomeSomeDescription"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-186xcr5"
                          >
                            <span>
                              Descripcion Subcuenta
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`ViewAccountPage Should render an Error dialog when the Account cant be found 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-145dpyk-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-acap47-MuiAlert-message"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
        >
          Oops!
        </div>
        Ha habido un 
        <strong>
          error
        </strong>
         al momento de intentar de cargar la cuenta con id. Favor de intentar mas tarde
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewAccountPage Should render the Account 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqvsg-MuiPaper-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h5 css-1hxkzy9-MuiTypography-root"
      >
        Vista de Cuenta
      </h4>
      <form
        id="create-account"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="title"
                id="title-label"
              >
                Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="AccountTitleId"
                  disabled=""
                  id="title"
                  name="title"
                  type="text"
                  value="AnAcct"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Tipo de Cuenta
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  aria-disabled="true"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="accountType"
                  class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="accountType"
                  role="button"
                >
                  Gastos
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  disabled=""
                  name="accountType"
                  tabindex="-1"
                  value="Expenses"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-186xcr5"
                  >
                    <span>
                      Tipo de Cuenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Subcuentas
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Id de Subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  >
                    Nombre de subcuenta
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                        for="description"
                        id="description-label"
                      >
                        Descripcion Subcuenta
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          data-testid="ATextField"
                          disabled=""
                          id="description"
                          name="description"
                          type="text"
                          value="SomeSomeDescription"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-186xcr5"
                          >
                            <span>
                              Descripcion Subcuenta
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1yhpg23-MuiTableCell-root"
                  />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
